consumer-java/aws-kinesis-video-producer-sdk-canary-consumer/src/main/java/com/amazon/kinesis/video/canary/consumer/CanaryFrameProcessor.java

```diff
@@ -61,8 +65,14 @@ public void process(Frame frame, MkvTrackMetadata trackMetadata, Optional<Fragme
                 .withMetricName("FrameSizeMatch")
                 .withUnit(StandardUnit.None)
                 .withValue(frameSize == data.length ? 1.0 : 0)
-                .withDimensions(dimension);
+                .withDimensions(dimensionPerStream);
```

Comment:
Why not just adding aggregatedDimension instead of creating a new datum? Same for the rest.

Reply:
That leads to creating a dimension of the format `ProducerSDKCanaryStreamName, ProducerSDKCanaryType`
I am looking to have two separate dimensions `ProducerSDKCanaryStreamName` and `ProducerSDKCanaryType`

Reply:
Talked offline. It makes sense why we need two separate datums.